### PR TITLE
Show private post filter only if private post exists

### LINKED ISSUE #265: Show private post filter only if private post exists

### PR DESCRIPTION
**Pull request by rajanit2000**
Fix #265

### CODE REVIEW

File: include/ui/form.php [hunk 1]
```diff
@@ -95,14 +95,20 @@ function bd_render_sticky_settings( $slug ) {
  * @param string $slug The slug to be used in field names.
  */
 function bd_render_private_post_settings( $slug ) {
-?>
+	$args  = array(
```
Comment (sudar) [inline include/ui/form.php]: Move this to a seperate function called `are_private_posts_present`.

Also this doesn't handle private posts that belong to a custom post type.

Comment (rajanit2000) [reply include/ui/form.php]: Fixed @sudar

File: include/ui/form.php [hunk 2]
```diff
@@ -87,6 +87,26 @@ function bd_render_sticky_settings( $slug ) {
...
 }
 
+/**
+ * Check there is have any private post is exist.
+ *
+ * @since 6.0.0
+ *
+ * @param string $post_type.
+ */
+function are_private_posts_present( $post_type='any') {
```
Comment (sudar) [inline include/ui/form.php]: Move this function to `Renderer` class.

File: include/ui/form.php [hunk 3]
```diff
@@ -87,6 +87,26 @@ function bd_render_sticky_settings( $slug ) {
...
+	$args  = array(
+		'post_status' => array( 'private' ),
+		'post_type'   => $post_type,
+	);
+	$query = new WP_Query( $args );
+	if ( $query->have_posts() ) {
```
Comment (sudar) [inline include/ui/form.php]: Replace the below statements with just `return $query->have_posts()`

Comment (rajanit2000) [reply include/ui/form.php]: @sudar fixed

**Pull request merged**